src/client/VZSidebar/DeleteConfirmationModal.tsx

```diff
@@ -1,36 +1,47 @@
 import { Button, Modal } from '../bootstrap';
 
-// Confirmation modal for deleting a file or directory.
-// Inspired by https://react-bootstrap.netlify.app/docs/components/modal/#live-demo
 export const DeleteConfirmationModal = ({
   show,
   onClose,
   onConfirm,
   isDirectory,
   name,
-}) => (
-  <Modal show={show} onHide={onClose} centered>
-    <Modal.Header closeButton onClick={onClose}>
-      <Modal.Title>
-        Delete {isDirectory ? 'Directory' : 'File'}
-      </Modal.Title>
-    </Modal.Header>
+}) => {
+  const handleSubmit = (event) => {
```

Comment:
Please leverage `useCallback` here, otherwise it's not very efficient (event listener gets added and removed each render). Thanks!

See https://react.dev/reference/react/useCallback

Reply:
Leveraged in handleSubmit